cluster/gce/gci/configure-helper-testcase.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gci
+
+import (
+	"github.com/golang/glog"
+
+	goruntime "runtime"
+
+	"testing"
+
+	"text/template"
+
+	"io/ioutil"
+
+	"k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"os"
+	"path/filepath"
+
+	"fmt"
+	"io"
+	"os/exec"
+	"path"
+
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+)
+
+const (
+	envScriptFileName         = "kube-env"
+	configureHelperScriptName = "configure-helper.sh"
+)
+
+type ManifestTestCase struct {
+	pod                 v1.Pod
+	envScriptPath       string
+	manifest            string
+	auxManifests        []string
+	kubeHome            string
+	manifestSources     string
+	manifestDestination string
+	manifestTemplate    string
+	manifestFuncName    string
+	manifestTemplateDir string
+	t                   *testing.T
+}
+
+func NewManifestTestCase(manifestUnderTest, funcUnderTest string, auxManifests []string, t *testing.T) *ManifestTestCase {
+	c := &ManifestTestCase{}
+	c.t = t
+	d, err := ioutil.TempDir("/tmp", "")
+	if err != nil {
+		c.t.Fatalf("Failed to create temp directory: %v", err)
+	}
+
+	c.kubeHome = d
+	glog.V(3).Infof("KUBE_HOME of the test case is %s", c.kubeHome)
```

Comment:
Tests should generally not log.
If there's something that would help debug the test, use t.Logf.
In any case, you can Log("test case: %+v", c) at the end of this func, instead of printing individual fields.

Reply:
Done.